src/prpy/base/wam.py

```diff
@@ -292,17 +293,22 @@ def MoveUntilTouch(manipulator, direction, distance, max_distance=None,
         # Execute on the real robot by tagging the trajectory with options that
         # tell the controller to stop on force/torque input.
         if not manipulator.simulated:
-            manipulator.SetTrajectoryExecutionOptions(traj, stop_on_ft=True,
+            manipulator.SetTrajectoryExecutionOptions(path, stop_on_ft=True,
                 force_direction=force_direction, force_magnitude=max_force,
                 torque=max_torque)
 
             manipulator.hand.TareForceTorqueSensor()
 
+            vl = manipulator.GetVelocityLimits()
```

Comment:
We should replace this manual saving logic with a `KinBodyStateSaver` with the `Save_JointMaxVelocityAndAcceleration` option. We should also make this scale factor (currently 0.25)` a parameter.